src/metabase/query_processor/streaming.clj

```diff
@@ -35,14 +35,15 @@
        (mbql.u/uniquify-names (map :name cols))))
 
 (defn- validate-table-columms
-  "Validate that all of the field refs in `table-columns` correspond to actual columns in `cols`, if `cols` contains
-  field refs. Returns `nil` if any do not, so that we fall back to using `cols` directly for the export (#19465).
+  "Validate that all of the columns in `table-columns` correspond to actual columns in `cols`, correlating them by
+  field ref or name. Returns `nil` if any do not, so that we fall back to using `cols` directly for the export (#19465).
   Otherwise returns `table-columns`."
   [table-columns cols]
-  (let [col-field-refs (set (remove nil? (map :field_ref cols)))]
-    ;; If there are no field refs in `cols` (e.g. for native queries), we should use `table-columns` as-is
-    (when (or (empty? col-field-refs)
```

Comment:
Removed this check, which was a special case for native queries that don't have field refs in their result metadata here, since we can correlate by name instead.